### ISSUE TITLE
Cocoapods: duplicated headers

### ISSUE THREAD

**Issue by alistra** [mixpanel/mixpanel-iphone#976]
**Integration Method:** CocoaPods
**Xcode Version:** 13.2.1
**Library Version:** 4.1.0
**Platform:** iOS
**Language:** Swift
**Description:** 

When integrating with Cocoapods, there are warnings related to duplicate headers in build phase and in fact the headers are duplicated there.

<img width="1174" alt="Screenshot 2022-01-19 at 13 34 35" src="https://user-images.githubusercontent.com/233144/150132034-7bea9c19-3b2a-4230-9ca0-ed9597a0dce2.png">

<img width="550" alt="Screenshot 2022-01-18 at 16 02 01" src="https://user-images.githubusercontent.com/233144/150132150-3f281d99-3cc5-466b-a0c5-7462662c3d4e.png">

**Expected Behavior:** 

There should be no warnings.

**Comment by oronbz** [comment]
Yeah, I'm also getting `[!] [Xcodeproj] Generated duplicate UUIDs:` after pod install, with the following duplicates:

```
AutomaticEvents.h
AutomaticEvents.m
MPDB.h
MPDB.m
MPFoundation.h
MPJSONHander.h
MPJSONHandler.m
MPLogger.h
MPLogger.m
MPNetwork.h
MPNetwork.m
MPNetworkPrivate.h
Mixpanel.h
Mixpanel.m
MixpanelExceptionHandler.h
MixpanelExceptionHandler.m
MixpanelGroup.h
MixpanelGroup.m
MixpanelGroupPrivate.h
MixpanelIdentity.h
MixpanelIdentity.m
MixpanelPeople.h
MixpanelPeople.m
MixpanelPeoplePrivate.h
MixpanelPersistence.h
MixpanelPersistence.m
MixpanelPrivate.h
MixpanelType.h
MixpanelType.m
SessionMetadata.h
SessionMetadata.m
c.h
umbrella.h
```

Version: 4.1.0

**Comment by lhasiuk** [comment]
Integration Method: CocoaPods 1.11.2
Xcode Version: 13.2.1
Library Version: 4.1.0
Platform: iOS
Language: Objective-C

I get the following warnings when building our app:
Skipping duplicate build file in Copy Headers build phase: /Users/lee/Desktop/<app directory name redacted>/Pods/Mixpanel/Sources/Mixpanel.h
Skipping duplicate build file in Copy Headers build phase: /Users/lee/Desktop/<app directory name redacted>/Pods/Mixpanel/Sources/MixpanelGroup.h
Skipping duplicate build file in Copy Headers build phase: /Users/lee/Desktop/<app directory name redacted>/Pods/Mixpanel/Sources/MixpanelPeople.h
Skipping duplicate build file in Copy Headers build phase: /Users/lee/Desktop/<app directory name redacted>/Pods/Mixpanel/Sources/MixpanelType.h

The Xcode recommended changes contain a section to correct this (as shown on the screen capture, below). That works until the next "pod install" is run, then the errors come back.

 
![Screen Shot 2022-01-20 at 7 24 38 PM](https://user-images.githubusercontent.com/479619/150443452-931e2bf8-c864-4872-91e5-b9f670cb7f48.png)

**Comment by jaredmixpanel** [comment]
@alistra @oronbz @lhasiuk thanks, we will look into this and get it fixed

**Comment by zihejia** [comment]
hi @alistra, this has been fixed in v4.1.2. Thanks for bringing it up. I'm closing it now, please feel free to reopen if the problem still exists.
